project-management/Github.md

```diff
@@ -0,0 +1,77 @@
+# Managing public (or private) projects using GitHub
+
+We use a common set of labels across projects in GitHub. It is
+possible to precisely control this set of labels using
+the [settings GitHub app][github-app-settings] (needs to be enabled
+for your org). Use the [`settings.yml`](../.github/settings.yml) from
+this repository as a starting point.
+
+[github-app-settings]: https://github.com/probot/settings
+
+## Labels
+
+We use labels to classify tasks by priority, from P0 to P4:
+
+* **P0 (blocker)** Do not do anything else, this is a blocker.
+* **P1 (critical)** Must have.
+* **P2 (major)** Should have.
+* **P3 (minor)** Could have.
+* **P4 (trivial)** Someday/maybe. Consider closing as "wontfix".
+
+Labels are also useful to classify issues by type or by component. Use
+colours to distinguish whether a label designates a type, a component
+or something else.
+
+## Project leads
+
+Each project must have a project lead (maintainer). By convention, the
+PL is whoever appears at the top of [`CODEOWNERS`][github-codeowners]
+file, e.g.:
+
+```
+* @mboes
+```
+
+This has the effect that @mboes will automatically be tagged as
+a reviewer for all incoming PR's, unless later lines in the same file
+override that.
+
+[github-codeowners]: https://help.github.com/articles/about-codeowners/
+
+## Milestones
+
+We gate tickets with milestones. Milestones always have an associated
+due date.
+
+## Releases
+
+Every public release must include a `CHANGELOG.md` file, in the format
+of [keepachangelog]. New entries to the changelog are integral parts
+of bug fix and feature PR's.
```

Comment:
If we're using the standardised form of CHANGELOG (and, I'm all for it). Should we consider looking into automated CHANGELOG generators at the same time?

If not, I think this guide should have some guidelines on what we expect to see in a CHANGELOG.

Reply:
I'm not sure I understand. This is a very simple format, and the guidelines are in the link already mentioned in the draft. Are you saying you want to copy these guidelines inline?

Reply:
No, the keep-a-changelog website is very good, and we just can refer to it. But it speaks in very general terms. It is useful to have an instantiation of the rules in terms of our workflow. Like how do I recognise a PR which deserves an entry in the CHANGELOG? Should we link to PR numbers in the changelog? To issue numbers? Should we give attribution in the changelog? 

These guidelines don't need to be complete: we can also address questions as they go. But having a set of guidelines already set help ask the right question and edit the document with the answers.

Reply:
I'm sure that we'll figure out some guidelines as we go. I do like inserting PR numbers (though it's a tad annoying as it's a two-step process: we have first to create the PR, then update the PR to refer to its own number).

I think tools like [this](https://github.com/clog-tool/clog-cli) link to PRs, too. I've been willing to give them a try for a while, but never did. So I have no data.

Another possible policy is: whenever (but not necessarily exclusively) a PR closes an issue, it should make it into the changelog, with a link to the issue.

Reply:
PR links are painful, for the reason you mention, and arguably not very helpful. The good news is that users are interested in tracking issues, not PR's (which are just a, sometimes partial, resolution to the issue). The issue itself gets updated with a link to the PR('s) whenever a change relevant to the issue makes it into mainline. So I'm adding this:

> Always try to link entries to issue numbers if a relevant issue exists. It's best to link issues as follows:
> 
> ```
> * Frobnicated knobs. See [#NNN](https://github.com/foocorp/bar/issues/NNN)
> ```

`clog-cli` looks very good. In a future version of this document, it's plausible that we'll recommend `clog`, if using it pans out.